omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+            acceleration_scale=1.0,
+            self_collision_check=True,
+            # maximum_trajectory_dt=0.25, #2.0, #0.25,
+            fixed_iters_trajopt=True,
+            finetune_trajopt_iters=100,
+            finetune_dt_scale=1.05,
+            velocity_scale=[1.0] * robot.n_joints,  # [0.25, 1, 1, 1, 1.0, 1.0, 1.0, 1.0, 1.0],
+        )
+        if motion_cfg_kwargs is not None:
+            motion_kwargs.update(motion_cfg_kwargs)
+        motion_gen_config = lazy.curobo.wrap.reacher.motion_gen.MotionGenConfig.load_from_robot_config(
+            robot_cfg,
+            lazy.curobo.geom.types.WorldConfig(),  # world_cfg,
+            self._tensor_args,
+            store_trajopt_debug=self.debug,
+            **motion_kwargs,
+        )
+        self.mg = lazy.curobo.wrap.reacher.motion_gen.MotionGen(motion_gen_config)
+        # print("CuRobo warming up...")
+        # self.mg.warmup(enable_graph=False, warmup_js_trajopt=False, batch=batch_size)
+        # print("CuRobo is ready.")
+
+        # Store internal variables
+        self.robot = robot
+        self.ee_link = robot_cfg["kinematics"]["ee_link"]
+        self._fk = FKSolver(self.robot.robot_arm_descriptor_yamls[robot.default_arm], self.robot.urdf_path)
+        self._usd_help = lazy.curobo.util.usd_helper.UsdHelper()
+        self._usd_help.stage = og.sim.stage
+        self._ignore_collision_paths = [] if ignore_collision_paths is None else ignore_collision_paths
+        assert batch_size >= 2, f"batch_size must be >= 2! Got: {batch_size}"
+        self.batch_size = batch_size
+
+        # Initialize other variables modified at runtime
+        self._ignore_paths = None
+
+    def update_obstacles(self):
+        """
+        Updates internal world collision cache representation based on sim state
+        """
+        print("Updating CuRobo world, reading w.r.t.", self.robot.prim_path)
+        # Infer how many collision bodies there are in sim, and add 10 as an additional buffer
+        n_cols = 0
+
+        # Ignore all container metalinks
+        self._ignore_paths = ["container"]
+
+        # Ignore any visual only objects and any objects not part of the robot's current scene
+        for idx, scene in enumerate(og.sim.scenes):
+            if idx == self.robot.scene.idx:
+                for obj in scene.objects:
+                    if obj.visual_only:
+                        self._ignore_paths.append(obj.prim_path)
+                        continue
+                    for link in obj.links.values():
+                        n_cols += len(link.collision_meshes)
+            else:
+                self._ignore_paths.append(scene.prim_path)
+
+        # # TODO: This should be modified dynamically but it breaks )):
+        # assert isinstance(self.mg.world_coll_checker, WorldMeshCollision)
+        # self.mg.world_coll_checker.clear_cache()
+        # self.mg.world_coll_checker.create_collision_cache(mesh_cache=n_cols + 10)
```

Comment:
hmm why should these be modified dynamically? Because of the creation / destruction of objects during simulation (e.g. transition rules)?

Reply:
removed